File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/DevMenuUtils.java
package com.swmansion.reanimated;

import com.facebook.react.ReactApplication;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.devsupport.interfaces.DevOptionHandler;
import com.facebook.react.devsupport.interfaces.DevSupportManager;

public class DevMenuUtils {

  public static void addDevMenuOption(ReactApplicationContext context, DevOptionHandler handler) {
    // In Expo, `ApplicationContext` is not an instance of `ReactApplication`
    if (context.getApplicationContext() instanceof ReactApplication) {
      DevSupportManager devSupportManager =
          ((ReactApplication) context.getApplicationContext())
              .getReactHost()
              .getDevSupportManager();

      if (devSupportManager != null) {
        devSupportManager.addCustomDevOption("Toggle slow animations (Reanimated)", handler);
      } else {
        throw new RuntimeException("[Reanimated] DevSupportManager is not available");
      }
    }
  }
}

File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/DevMenuUtils.kt
package com.swmansion.reanimated

import com.facebook.react.ReactApplication
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.devsupport.interfaces.DevOptionHandler

object DevMenuUtils {
  @JvmStatic
  fun addDevMenuOption(context: ReactApplicationContext, handler: DevOptionHandler) {
    // In Expo, `ApplicationContext` is not an instance of `ReactApplication`
    if (context.applicationContext is ReactApplication) {
      val devSupportManager =
          (context.applicationContext as ReactApplication).reactHost!!.devSupportManager

      if (devSupportManager != null) {
        devSupportManager.addCustomDevOption("Toggle slow animations (Reanimated)", handler)
      } else {
        throw RuntimeException("[Reanimated] DevSupportManager is not available")
      }
    }
  }
}
